mito-ai/mito_ai/rules/handlers.py

```diff
@@ -0,0 +1,41 @@
+from dataclasses import dataclass
+import json
+from typing import Any, Final
+import tornado
+import os
+from jupyter_server.base.handlers import APIHandler
+from mito_ai.rules.utils import RULES_DIR_PATH, get_all_rules, get_rule, set_rules_file
+
+
+class RulesHandler(APIHandler):
+    """Handler for operations on a specific setting"""
+    
+    @tornado.web.authenticated
+    def get(self, key=None):
+        """Get a specific rule by key or all rules if no key provided"""
+        if key is None or key == '':
+            # No key provided, return all rules
+            rules = get_all_rules()
+            self.finish(json.dumps(rules))
+        else:
+            # Key provided, return specific rule
+            rule_content = get_rule(key)
+            if rule_content is None:
+                self.set_status(404)
+                self.finish(json.dumps({"error": f"Rule with key '{key}' not found"}))
+            else:
+                self.finish(json.dumps({"key": key, "content": rule_content}))
+    
+    @tornado.web.authenticated
+    def put(self, key):
+        """Update or create a specific setting"""
+        data = json.loads(self.request.body)
+        if 'content' not in data:
+            self.set_status(400)
+            self.finish(json.dumps({"error": "Content is required"}))
+            return
+            
+        set_rules_file(key, data['content'])
+        self.finish(json.dumps({"status": "updated", "rules file ": key}))
+
```

Comment:
Wouldn't it make sense to also add delete method ? something like this?


```suggestion
@tornado.web.authenticated
def delete(self, key):
    """Delete a specific rule"""
    try:
        if not key:
            return self._handle_error(400, "Rule key is required")
        
        if delete_rule(key):  # Assuming this function exists
            self._handle_success({"status": "deleted", "key": key})
        else:
            self._handle_error(404, f"Rule '{key}' not found")
    except Exception as e:
        self._handle_error(500, str(e))
```


Or is the expectation that the rules page dictates what rules to be added and the deletion of rules would be happening only in that page by the user

Reply:
I'm going to leave off DELETE for now because deleting files on the user's file system is always a little bit more dangerous incase the path gets fucked up and we accidentally delete something we don't want to be deleting. I've seen this happen in other Open Source Jupyter Extension.